pkg/vz/vm_darwin.go

```diff
@@ -239,6 +241,21 @@ func attachPlatformConfig(driver *driver.BaseDriver, vmConfig *vz.VirtualMachine
 	if err != nil {
 		return err
 	}
+
+	// nested virt
+	if *driver.Yaml.NestedVirtualization {
+		macOSProductVersion, err := osutil.ProductVersion()
+		if err != nil {
+			return fmt.Errorf("failed to get macOS product version: %w", err)
+		}
+
+		if macOSProductVersion.LessThan(*semver.New("15.0.0")) {
+			return fmt.Errorf("nested virtualization requires macOS 15 or newer")
+		}
```

Comment:
It will be little nicer to add a helper like isNestedVirtualiztionAvailable() with these checks. This can also be implemented for linux, so it can be a common interface for all platforms.